comp/dogstatsd/server/parse.go

```diff
@@ -238,8 +246,25 @@ func (p *parser) parseFloat64List(rawFloats []byte) ([]float64, error) {
 }
 
 // extractContainerID parses the value of the container ID field.
+// If the field is prefixed by `in-`, it corresponds to the cgroup controller's inode of the source
+// and is used for ContainerID resolution.
 func (p *parser) extractContainerID(rawContainerIDField []byte) []byte {
-	return rawContainerIDField[len(containerIDFieldPrefix):]
+	containerIDField := rawContainerIDField[len(containerIDFieldPrefix):]
+
+	if bytes.HasPrefix(containerIDField[:len(containerIDFieldInodePrefix)], containerIDFieldInodePrefix) {
+		inodeField, err := strconv.ParseUint(string(containerIDField[len(containerIDFieldPrefix)+1:]), 10, 64)
+		if err != nil {
+			return nil
```

Comment:
Is it worth logging a trace log? or even a debug one?
When troubleshooting, having no information that a malformed inode has been transmitted could be painful.

Reply:
There is also a telemetry for "origin detection errors" in `pkg/tagger/global.go` which already has a `log.Tracef`. This may be sufficient.

Lets confirm that we see this trace and increment the telemetry on error though https://github.com/DataDog/datadog-agent/blob/41126b6475ed4314198440e9283af71b48e107b1/pkg/tagger/global.go#L58-L59

(I'm not sure if this condition will trigger the `tagger` codepath or not)

Reply:
Got it, I've added a debug log.
```
2023-12-15 09:44:43 UTC | CORE | DEBUG | (comp/dogstatsd/server/parse.go:258 in extractContainerID) | Failed to parse inode from in-NotAnInode, got strconv.ParseUint: parsing "NotAnInode": invalid syntax
```